src/components/OrdersWidget/index.tsx

```diff
@@ -44,49 +54,104 @@ const OrdersWidget: React.FC = () => {
   // this page is behind login wall so networkId should always be set
   const { networkId } = useWalletConnection()
 
-  const [orders, setOrders] = useSafeState<AuctionElement[]>(allOrders)
-  const [showActive, setShowActive] = useSafeState<boolean>(true)
+  // allOrders and markedForDeletion, split by tab
+  const [filteredOrders, setFilteredOrders] = useSafeState<FilteredOrdersState>(emptyState())
+  const [selectedTab, setSelectedTab] = useSafeState<OrderTabs>('active')
+
+  // syntactic sugar
+  const { displayedOrders, markedForDeletion } = useMemo(
+    () => ({
+      displayedOrders: filteredOrders[selectedTab].orders,
+      markedForDeletion: filteredOrders[selectedTab].markedForDeletion,
+    }),
+    [filteredOrders, selectedTab],
+  )
+
+  const setSelectedTabFactory = useCallback(
+    (type: OrderTabs): ((event: React.SyntheticEvent<HTMLButtonElement | HTMLFormElement>) => void) => (
+      event: React.SyntheticEvent<HTMLButtonElement | HTMLFormElement>,
+    ): void => {
+      // form is being submitted when clicking on tab buttons, thus preventing default
+      event.preventDefault()
 
-  const toggleShowActive = useCallback(() => {
-    setShowActive(isActive => !isActive)
-  }, [setShowActive])
+      setSelectedTab(type)
+    },
+    [setSelectedTab],
+  )
 
+  // Update filteredOrders state whenever there's a change to allOrders
+  // splitting orders into respective tabs
   useEffect(() => {
     const now = new Date()
-    const filtered = allOrders.filter(order => showActive === isOrderActive(order, now))
-    setOrders(filtered)
-  }, [allOrders, setOrders, showActive])
 
-  const shownOrdersCount = orders.length
+    const filteredOrders = emptyState()
```

Comment:
Leave it like this. I think it's good. But an alternative would be to use `reduce` starting on an empty state